flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskStopTest.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.runtime.taskmanager;
+
+import java.lang.reflect.Field;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.runtime.broadcast.BroadcastVariableManager;
+import org.apache.flink.runtime.deployment.TaskDeploymentDescriptor;
+import org.apache.flink.runtime.execution.ExecutionState;
+import org.apache.flink.runtime.execution.librarycache.LibraryCacheManager;
+import org.apache.flink.runtime.executiongraph.ExecutionAttemptID;
+import org.apache.flink.runtime.filecache.FileCache;
+import org.apache.flink.runtime.instance.ActorGateway;
+import org.apache.flink.runtime.io.disk.iomanager.IOManager;
+import org.apache.flink.runtime.io.network.NetworkEnvironment;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.jobgraph.tasks.AbstractInvokable;
+import org.apache.flink.runtime.jobgraph.tasks.Stoppable;
+import org.apache.flink.runtime.memory.MemoryManager;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import scala.concurrent.duration.FiniteDuration;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ TaskDeploymentDescriptor.class, JobID.class, FiniteDuration.class })
+public class TaskStopTest {
+	private AbstractInvokable taskMock;
+	private Task task;
+
+	public void doMocking() throws Exception {
+		TaskDeploymentDescriptor tddMock = mock(TaskDeploymentDescriptor.class);
+		when(tddMock.getNumberOfSubtasks()).thenReturn(1);
+		when(tddMock.getJobID()).thenReturn(mock(JobID.class));
+		when(tddMock.getVertexID()).thenReturn(mock(JobVertexID.class));
+		when(tddMock.getExecutionId()).thenReturn(mock(ExecutionAttemptID.class));
+		when(tddMock.getTaskName()).thenReturn("taskName");
+		when(tddMock.getJobConfiguration()).thenReturn(mock(Configuration.class));
+		when(tddMock.getTaskConfiguration()).thenReturn(mock(Configuration.class));
+		when(tddMock.getInvokableClassName()).thenReturn("className");
+
+		task = new Task(tddMock, mock(MemoryManager.class), mock(IOManager.class), mock(NetworkEnvironment.class),
+				mock(BroadcastVariableManager.class), mock(ActorGateway.class), mock(ActorGateway.class),
+				mock(FiniteDuration.class), mock(LibraryCacheManager.class), mock(FileCache.class),
+				mock(TaskManagerRuntimeInfo.class));
+		Field f = task.getClass().getDeclaredField("invokable");
+		f.setAccessible(true);
+		f.set(task, taskMock);
```

Comment:
private member `Task.invokable` will be `null` until it is created and set in `Task.run()` which is never called in this tests.